core/src/allocator.zig

```diff
@@ -0,0 +1,614 @@
+const std = @import("std");
+const microzig = @import("microzig.zig");
+
+const Alignment = std.mem.Alignment;
+
+extern var microzig_heap_start: u8;
+extern var microzig_heap_end: u8;
+
+/// The number of free lists we maintain.
+const free_list_count: usize = 6;
+
+const Alloc = @This();
+
+free_lists: [free_list_count]?*Chunk = @splat(null),
+
+low_boundary: usize,
+high_boundary: usize,
+
+mutex: microzig.interrupt.Mutex = .{},
+
+/// Set up an allocator by adding all the memory that is not otherwise used by the program.
+/// It relies on the linker script to define the microzig_heap_start and microzig_heap_end
+/// symbols.
+///
+/// The allocator provided a `reserve` parameter to allow the user to reserve a small amount of
+/// memory at the end of the heap.  This prevents the allocator from using memory that is reserved
+/// for other purposes, such as the stack.
+///
+/// Example of use:
+/// ```
+/// // Get a heap allocator instance reserving 1024 bytes for the stack.
+/// var heap_allocator = microzig.core.Allocator.setupWithHeap(1024);
+///
+/// // Get the std.mem.Allocator from the heap allocator.
+/// const allocator : std.mem.Allocator = heap_allocator.allocator();
+/// ```
+///
+/// aParameters:
+/// - `reserve`: The number of bytes to omit at the end of the heap.
+///
+pub fn setupWithHeap(reserve: usize) Alloc {
```

Comment:
Use snake case for functions, also zig conventions to use the word init, so something like `init_with_heap`

Reply:
Done